src/app/components/checkin/allCheckIns.controller.js

```diff
@@ -5,23 +5,103 @@
     .module('app.checkin')
     .controller('AllCheckInsCtrl', AllCheckInsCtrl);
 
-  AllCheckInsCtrl.$inject = ['$scope', 'alertService', 'config', 'UserCheckInService', 'UserDataService', 'gettextCatalog'];
-  
-  function AllCheckInsCtrl ($scope, alertService, config, UserCheckInService, UserDataService, gettextCatalog) {
+  AllCheckInsCtrl.$inject = ['$exceptionHandler', '$scope', '$uibModal', 'alertService', 'config', 'UserCheckInService', 'UserDataService', 'gettextCatalog'];
+
+  function AllCheckInsCtrl ($exceptionHandler, $scope, $uibModal, alertService, config, UserCheckInService, UserDataService, gettextCatalog) {
     $scope.listsMember = [];
     $scope.leaveList = leaveList;
+    $scope.editCheckIn = editCheckIn;
+    $scope.showDatePicker = showDatePicker;
+    $scope.updateCheckIn = updateCheckIn;
+    $scope.closeEditModal = closeEditModal;
+    $scope.removeCheckOutDate = removeCheckOutDate;
     $scope.page = 1;
     $scope.itemsPerPage = 10;
     $scope.listSearchTerm = {
       name: ''
     };
+    $scope.datePicker = {
+      opened: false
+    };
+    $scope.dateOptions = {
+      maxDate: moment().add(5, 'year')._d,
+      minDate: new Date(),
+      showWeeks: false,
+      startingDay: 1
+    };
+    var editModal;
+
+    function showDatePicker () {
+      $scope.datePicker.opened = true;
+    }
+
+    function saveCheckIn (checkIn, checkInUser) {
+      UserCheckInService.update({userId: $scope.currentUser._id, listType: checkIn.type + 's', checkInId: checkIn._id}, checkInUser, function (user) {
+        $scope.setCurrentUser(user);
+        angular.forEach($scope.listsMember, function (list) {
+          if (list._id === checkInUser.list) {
+            list.checkoutDate = checkInUser.checkoutDate;
+          }
+        });
+        editModal.close();
+      }, function (error) {
+        alertService.add('danger', gettextCatalog.getString('There was an error saving'));
+        $exceptionHandler(error, 'Update checkout date');
+      });
+    }
+
+    function removeCheckOutDate (checkIn) {
+      var checkInUser = {
+        list: checkIn._id,
+        checkoutDate: null
```

Comment:
@amyvbenson From a technical standpoint, it's ok to remove the checkout date. From a usability standpoint however, this is more for Yaelle to answer, but I think we should avoid checkins without a checkout date, unless the checkin is within an organization... Let's wait for Yaelle's answer on this before merging.